microprofile/opentracing-smallrye/src/main/java/org/wildfly/microprofile/opentracing/smallrye/WildFlyTracerFactory.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.microprofile.opentracing.smallrye;
+
+import io.opentracing.Tracer;
+import io.opentracing.noop.NoopTracerFactory;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.jboss.as.controller.capability.RuntimeCapability;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceName;
+
+/**
+ * Registry of all the tracers configuration registered in WildFly.
+ * @author Emmanuel Hugonnet (c) 2019 Red Hat, Inc.
+ */
+public class WildFlyTracerFactory {
+
+    private static final Map<ServiceName, Configuration> CONFIGURATIONS = new HashMap<>();
```

Comment:
This can be Map<String, Configuration> with null used for the compatibility tracer. That eliminates all the conversion from simple tracer name string -> RuntimeCapability -> ServiceName.